components/targets/source-code-artifact/component.yaml

```diff
@@ -0,0 +1,34 @@
+name: "source-code-artifact"
+description: "Downloads archived files in zip, tag or targz format from a plain http or s3 compatible API"
+type: "target"
+parameters:
+  # Must be a valid url like: https://github.com/aws/aws-sdk-go-v2/archive/refs/heads/main.zip
+  # Or a s3 compatible url like: s3://my-bucket/my-archive.tar or gs://my-bucket/my-archive.tar.gz
+  - name: "artifact_url"
+    type: "string"
+    value: "{{ .context.trigger.event.url }}"
+  - name: "artifact_reference"
+    type: "string"
+    value: "{{ .context.trigger.event.reference }}"
+  - name: "artifact_registry_region"
+    type: "string"
+    value: ""
+  - name: "artifact_access_key"
```

Comment:
these are weird parameters because in the case of S3 they are the access key ID and secret and in the case of the HTTP auth, username and token. I'm not sure how to name them to make this clear, maybe access_key_or_username? or maybe have completely separate sets of parameters?

Reply:
Yeah this is not fantastic

What about `auth_id` and `auth_secret`?

Reply:
Done